contracts/libs/MarginVault.sol

```diff
@@ -31,6 +31,8 @@ library MarginVault {
 
     // vault is a struct of 6 arrays that describe a position a user has, a user can have multiple vaults.
     struct Vault {
+        // address of the margin pool for vault
```

Comment:
IMO it is better to keep using vaultType instead of adding address pool in the vault structure, currently we have 2 vault types, we can use `3` as borrowable vault, and then do the checks if vault type is 3 when depositing collateral, forward it to the borrowbalePool

Reply:
changing the vault struct will mess up with all existing vault in the storage right? or is it safe because all existing properties are arrays so they're not stored based on order?

Reply:
Yes I will set vault type index 2 to borrowable pool. Upon second glance I realize this won't cause any problems with payoff calculation / liquidation logic in MarginCalculator because this pool will only be used for covered calls (not a naked vault) so the pool and naked vaults are mutually exclusive.